src/Adapter/Product/Repository/CombinationRepository.php

```diff
@@ -0,0 +1,294 @@
+use PrestaShop\PrestaShop\Adapter\Attribute\Repository\AttributeRepository;
+use PrestaShop\PrestaShop\Core\Domain\Language\ValueObject\LanguageId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Exception\CannotAddCombinationException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Exception\CannotDeleteCombinationException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Exception\CombinationConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Exception\CombinationNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\ValueObject\CombinationId;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShopException;
+
+/**
+ * Provides access to Combination data source
+ */
+class CombinationRepository extends AbstractObjectModelRepository
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    /**
+     * @var AttributeRepository
+     */
+    private $attributeRepository;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param AttributeRepository $attributeRepository
+     */
+    public function __construct(
+        Connection $connection,
+        string $dbPrefix,
+        AttributeRepository $attributeRepository
+    ) {
+        $this->connection = $connection;
+        $this->dbPrefix = $dbPrefix;
+        $this->attributeRepository = $attributeRepository;
+    }
+
+    /**
+     * @param Combination $combination
+     *
+     * @return CombinationId
+     *
+     * @throws CombinationConstraintException
+     * @throws CoreException
+     */
+    public function add(Combination $combination): CombinationId
+    {
+        //@todo: validation?
+        $id = $this->addObjectModel($combination, CannotAddCombinationException::class);
+
+        return new CombinationId($id);
+    }
+
+    /**
+     * @param Combination $combination
+     * @param int $errorCode
+     *
+     * @throws CoreException
+     */
+    public function delete(Combination $combination, int $errorCode = 0): void
+    {
+        $this->deleteObjectModel($combination, CannotDeleteCombinationException::class, $errorCode);
+    }
+
+    /**
+     * @param ProductId $productId
+     * @param int $limit
+     * @param int $offset
+     * @param array $filters
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    public function getProductCombinations(ProductId $productId, ?int $limit = null, ?int $offset = null, array $filters = []): array
+    {
+        $qb = $this->getCombinationsQueryBuilder($productId, $filters)
+            ->select('pa.*')
+        ;
+
+        if ($offset) {
+            $qb->setFirstResult($offset);
+        }
+
+        if ($limit) {
+            $qb->setMaxResults($limit);
+        }
```

Comment:
```suggestion
        $qb = $this->getCombinationsQueryBuilder($productId, $filters)
            ->select('pa.*')
            ->setFirstResult($offset)
            ->setMaxResults($limit)
        ;
```

Reply:
I thought I answered this already :thinking: 
Anyway- $offset and $limit are optional params, I cannot set `null` neither as FirstResult nor as MaxResults

Reply:
`setFirstResult` & `setMaxResults` accept null or integer in parameter (:warning: it's not defined in PHPDoc :stuck_out_tongue:)

Reply:
hmm well yeah, because its not typed, but in theory its not what method expects :thinking:
Anyway, ok lets use this :smile: